examples/fluids/navierstokes.c

```diff
@@ -280,26 +281,50 @@ int main(int argc, char **argv) {
                           "  Mesh:\n"
                           "    Number of 1D Basis Nodes (P)       : %" CeedInt_FMT "\n"
                           "    Number of 1D Quadrature Points (Q) : %" CeedInt_FMT "\n"
                           "    Global DoFs                        : %" PetscInt_FMT "\n"
-                          "    Owned DoFs                         : %" PetscInt_FMT "\n"
                           "    DoFs per node                      : %" PetscInt_FMT "\n"
-                          "    Global nodes (DoFs / %" PetscInt_FMT ")            : %" PetscInt_FMT "\n"
-                          "    Local nodes                        : %" PetscInt_FMT "\n",
-                          num_P, num_Q, glob_dofs, owned_dofs, num_comp_q, num_comp_q, glob_nodes, local_nodes));
+                          "    Global nodes                       : %" PetscInt_FMT "\n",
+                          num_P, num_Q, glob_dofs, num_comp_q, glob_dofs / num_comp_q));
+    // -- Get Partition Statistics
+    PetscInt sendbuf[2], part_owned_dofs[2], part_local_dofs[2];
+    sendbuf[0] = sendbuf[1] = owned_dofs;
+    PetscCall(PetscGlobalMinMaxInt(comm, sendbuf, part_owned_dofs));
+    sendbuf[0] = sendbuf[1] = local_dofs;
+    PetscCall(PetscGlobalMinMaxInt(comm, sendbuf, part_local_dofs));
+    PetscReal part_owned_dof_ratio = (PetscReal)part_owned_dofs[1] / (PetscReal)part_owned_dofs[0];
+    PetscReal part_local_dof_ratio = (PetscReal)part_local_dofs[1] / (PetscReal)part_local_dofs[0];
+    PetscCall(PetscPrintf(comm,
+                          "  Partition:\n"
+                          "    Owned free nodes (min,max,ratio)   : %" PetscInt_FMT ", %" PetscInt_FMT ", %f\n"
+                          "    Local nodes (min,max,ratio)        : %" PetscInt_FMT ", %" PetscInt_FMT ", %f\n"
+                          "(nodes == DoFs / %" PetscInt_FMT ")\n",
+                          part_owned_dofs[0] / num_comp_q, part_owned_dofs[1] / num_comp_q, part_owned_dof_ratio, part_local_dofs[0] / num_comp_q,
+                          part_local_dofs[1] / num_comp_q, part_local_dof_ratio, num_comp_q));
+    {
+      PetscSF            sf;
+      PetscInt           niranks;
+      const PetscInt    *ioffset, *irootloc;
+      const PetscMPIInt *iranks;
+      PetscCall(DMGetSectionSF(user->dm, &sf));
+      PetscCall(PetscSFGetLeafRanks(sf, &niranks, &iranks, &ioffset, &irootloc));
+      PetscInt shared_dofs = ioffset[niranks];
+      PetscInt part_shared_dofs[2];
+      sendbuf[0] = sendbuf[1] = shared_dofs;
+      PetscCall(PetscGlobalMinMaxInt(comm, sendbuf, part_shared_dofs));
+      PetscReal part_shared_dof_ratio = (PetscReal)part_shared_dofs[1] / (PetscReal)part_shared_dofs[0];
+      PetscCall(PetscPrintf(comm, "    Shared nodes (min,max,ratio)       : %" PetscInt_FMT ", %" PetscInt_FMT ", %f\n",
+                            part_shared_dofs[0] / num_comp_q, part_shared_dofs[1] / num_comp_q, part_shared_dof_ratio));
+    }
```

Comment:
Note that this currently doesn't work as expected. I'd think that the length of `irootloc` would be the number of remote leaves referencing local roots (which I'm fairly certain would be shared dofs), but the result of that has the values seemingly much too large:

```
 ❯ mpirun -n 2 ./navierstokes -options_file blasius.yaml -ts_monitor -snes_monitor -ts_max_steps 1 -petscpartitioner_view -dm_view -dm_plex_box_faces 41,61,1
Graph Partitioner: 2 MPI Processes
  type: simple
  edge cut: 0
  balance: 0
  use vertex weights: 1
DM Object: box 2 MPI processes
  type: plex
box in 3 dimensions:
  Number of 0-cells per rank: 2648 2646
  Number of 1-cells per rank: 6472 6467
  Number of 2-cells per rank: 5076 5072
  Number of 3-cells per rank: 1251 1250
Labels:
  depth: 4 strata with value/size (0 (2648), 1 (6472), 2 (5076), 3 (1251))
  marker: 1 strata with value/size (1 (10503))
  Face Sets: 5 strata with value/size (1 (1251), 2 (1251), 3 (41), 5 (30), 6 (31))
  celltype: 4 strata with value/size (0 (2648), 1 (6472), 4 (5076), 7 (1251))

-- Navier-Stokes solver - libCEED + PETSc --
  MPI:
    Host Name                          : fermat
    Total ranks                        : 2
  Problem:
    Problem Name                       : blasius
    Stabilization                      : SUPG
  libCEED:
    libCEED Backend                    : /cpu/self/avx/blocked
    libCEED Backend MemType            : host
  PETSc:
    Box Faces                          : 41,61,1
    DM MatType                         : aij
    DM VecType                         : standard
    Time Stepping Scheme               : implicit
  Mesh:
    Number of 1D Basis Nodes (P)       : 2
    Number of 1D Quadrature Points (Q) : 2
    Global DoFs                        : 20664
    DoFs per node                      : 5
    Global nodes                       : 4132
  Partition:
    Owned free nodes (min,max,ratio)   : 2016, 2116, 1.050000
    Local nodes (min,max,ratio)        : 2646, 2648, 1.000756
(nodes == DoFs / 5)
    Shared nodes (min,max,ratio)       : 2016, 2185, 1.084127
```
Does the section `SF` map *all* non-fixed dofs rather than just the "necessary" ones?

Reference: [`PetscSFGetLeafRanks`](https://petsc.org/release/manualpages/PetscSF/PetscSFGetLeafRanks/)

Reply:
Actually, I think I'm overlooking that the section `SF` maps global `Vec`s to local `Vec`s rather than just the remote nodes. So I should discount the `ioffset` value that corresponds to the local rank?